test/v2/updateUKSavingsAccountName/UpdateUKSavingsAccountNameControllerSpec.scala

```diff
@@ -0,0 +1,116 @@
+import shared.models.errors._
+import shared.models.outcomes.ResponseWrapper
+import v2.updateUKSavingsAccountName.fixture.UpdateUKSavingsAccountNameFixtures.{requestBodyModel, validRequestJson}
+import v2.updateUKSavingsAccountName.model.request.UpdateUKSavingsAccountNameRequest
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class UpdateUKSavingsAccountNameControllerSpec
+  extends ControllerBaseSpec
+    with ControllerTestRunner
+    with MockUpdateUKSavingsAccountNameService
+    with MockUpdateUKSavingsAccountNameValidatorFactory {
+
+  val savingsAccountId: String            = "SAVKB2UVwUTBQGJ"
+
+  private val requestData: UpdateUKSavingsAccountNameRequest = UpdateUKSavingsAccountNameRequest(
+    nino = parsedNino,
+    savingsAccountId = SavingsAccountId(savingsAccountId),
+    body = requestBodyModel
+  )
+
+  "UpdateUKSavingsAccountNameController" should {
+    "return a successful response with status 204 (NO_CONTENT)" when {
+      "given a valid request" in new Test {
+        willUseValidator(returningSuccess(requestData))
+
+        MockUpdateUKSavingsAccountNameService
+          .createAmend(requestData)
+          .returns(Future.successful(Right(ResponseWrapper(correlationId, ()))))
+
+        runOkTestWithAudit(expectedStatus = NO_CONTENT, maybeAuditRequestBody = Some(validRequestJson))
+      }
+    }
+
+    "return the error as per spec" when {
+      "the parser validation fails" in new Test {
+        willUseValidator(returning(NinoFormatError))
+
+        runErrorTestWithAudit(NinoFormatError, Some(validRequestJson))
+      }
+
+      "service returns an error" in new Test {
+        willUseValidator(returningSuccess(requestData))
+
+        MockUpdateUKSavingsAccountNameService
+          .createAmend(requestData)
+          .returns(Future.successful(Left(ErrorWrapper(correlationId, AccountNameFormatError))))
+
+        runErrorTestWithAudit(AccountNameFormatError, Some(validRequestJson))
+      }
+    }
+  }
+
+  private trait Test extends ControllerTest with AuditEventChecking[GenericAuditDetail] {
+
+    val controller: UpdateUKSavingsAccountNameController = new UpdateUKSavingsAccountNameController(
+      authService = mockEnrolmentsAuthService,
+      lookupService = mockMtdIdLookupService,
+      validatorFactory = mockUpdateUKSavingsAccountNameValidatorFactory,
+      service = mockUpdateUKSavingsAccountNameService,
+      auditService = mockAuditService,
+      cc = cc,
+      idGenerator = mockIdGenerator
+    )
+
+    MockedSharedAppConfig.featureSwitchConfig.anyNumberOfTimes() returns Configuration(
+      "supporting-agents-access-control.enabled" -> true
+    )
+
+    MockedSharedAppConfig.endpointAllowsSupportingAgents(controller.endpointName).anyNumberOfTimes() returns false
+
+    protected def callController(): Future[Result] = controller.updateUKSavingsAccountName(validNino, savingsAccountId)(fakePostRequest(validRequestJson))
```

Comment:
Using `fakePostRequest` here can be misleading as this is a PUT and we can have updates with POST. I suggest either updating here to `fakeRequest.withBody(validRequestJson)` or rename `fakePostRequest` to a generic name like `fakeRequestWithBody` but this change will affect other tests. This possibly affects other APIs also.